src/disclosure-manager/_disclosureCallbacks.js

```diff
@@ -0,0 +1,24 @@
+let disclosureCallbacks = [];
+
+const addCallback = (callback) => {
+  disclosureCallbacks = [...disclosureCallbacks, callback];
+};
+
+const removeCallback = (callback) => {
+  disclosureCallbacks = disclosureCallbacks.filter(disclosureCallback => disclosureCallback !== callback);
+};
+
+const closeMostRecentDisclosure = () => {
+  if (!disclosureCallbacks.length) {
+    return false;
+  }
+
+  disclosureCallbacks[disclosureCallbacks.length - 1]();
```

Comment:
Both goBack/closeDisclosure will return a Promise. The Promise will resolve if the dismissal happens, but it will be rejected if the user prevents the dismissal. I would return that promise here to give the caller the most flexibility.

 You could return a resolved promise directly if there's nothing in the stack. You could expose a `getStackSize` function or something if you think consumers of this will need to know that information.

Reply:
Updated to return a promise.  This will change a bit how the android integration will work.  Before I was planning on just blindly calling the dismiss function and notifying the Android shell if it returned false.

With these changes, "getActiveDisclosureCount" will be needed and will need to be checked before attempting to dismiss. If it returns 0 we can notify android instead of calling dismiss.